docs/preview/features/openapi/health-checks.md

```diff
@@ -0,0 +1,58 @@
+﻿---
+title: "Using health checks report in OpenAPI documentation"
+layout: default
+---
+
+# Using health checks report in OpenAPI documentation
+
+Microsoft has a way of providing the health status of an application by running over a series of 'health checks' that eventually generate a health checks report. 
+For more information on application health, see [Microsoft's documentation](https://docs.microsoft.com/en-us/aspnet/core/host-and-deploy/health-checks).
+
+The problem is that the health checks report model called `HealthReport` cannot be safely used as a JSON serializable type, as it could contain exception details (and therefore assembly information).
+Arcus provides a dedicated JSON data-trasfer object (DTO) to expose the health status of the application in a reliable and safe manner.
+
+## Installation
+
+This feature requires to install our NuGet package
+
+```shell
+PM > Install-Package Arcus.WebApi.OpenApi.Extensions
```

Comment:
I would 've added this to the Arcus.WebApi project itself, instead of adding a separate package.
IMHO, this is not only linked with the OpenAPI docs, it is instead actually the API model of the Health endpoint.

Reply:
It's not added to a new package, though, since we already have something for OpenAPI and since it was OpenAPI that was the cause that this problem bubbled up.
But see your point. What other place could we put it that isn't a new package? Maybe `Arcus.WebApi.Hosting`?

Reply:
I had to open Arcus.WebApi to see what we all had.   To be honest, I don't find a suitable place right away.

Maybe we should just add it as a class in the WebApi template project ?  After all, it is only there that it is being used.
When people start a new project based on the Arcus Template, they get a predefined health-endpoint.  The HealthReport API model belongs to that endpoint, so I would keep it close to where it is being used atm.
If it turns out later that we can put that class somewhere else, we can always change it.

Reply:
The `HealthReport` is actually also used in the TCP health probe in the messaging projects, so it would be a good thing that it's added there too.

Reply:
Yeah, we could. Was just wondering if this is maybe a bit too low-level. Not very user-friendly to just put it there. It's also one of the reasons we added MVC extensions for the JSON formatting, bc we didn't want too much additional code there for things that will always be the same kind of structure.

Reply:
I'm not sure what you mean here.

IMHO, the HealthReport class is a (api) model.  It belongs close to the API operation that uses / returns this model.  Therefore, I see no problem in putting it in the Arcus WebAPI template f.i.

Reply:
> I'm not sure what you mean here.

Recently, we updated the Web API template, bc it contained a lot of custom code (that was repeated in each new project creation from the template). This was a lot of duplication and made the overall user experience worse. We decided to extract the custom code to MVC extensions so that we can maintain the custom code better and the user experience is greatly improved. See this issue for more info: https://github.com/arcus-azure/arcus.webapi/issues/250
The reason why I mention this, is that we are now considering to again place custom code in three project templates and while the custom code of the MVC options was mostly configuration code, is the custom health report for serialization an even more 'internal' type. So, bit reluctant to add this just as-is to the template projects as it conflicts with the 'mindset' we were starting with the templates.

Reply:
Maybe, we could include the health report JSON dto to the `Arcus.Messaging.Health` package, and use it there. (not sure if we would qualify it as a breaking change there), as the API operation we set up is set up there.

Reply:
Is it really custom code ?  It is an API model and imo API models should live close to where they're used.
I don't think we should share the same model between multiple services, as this would mean that they cannot evolve separately.  
The HealthReport model is not infrastructure code, so to me it doesn't feel problematic to add it directly into the template.  Moreover, you allow users of the template to fine-tune / modify that model if they want to.

Reply:
What exactly do you mean with 'adding it explicitly' ?

Reply:
Adding it to the template, like proposed in: https://github.com/arcus-azure/arcus.templates/pull/589

Reply:
Added it here in the messaging library: https://github.com/arcus-azure/arcus.messaging/issues/287 we can close this PR.